frontend/utils/urdUtils.ts

```diff
@@ -0,0 +1,218 @@
+import { ethers } from 'ethers';
+import { getNetworkConfig } from '@/constants/networks';
+import UniversalProfile from '@lukso/lsp-smart-contracts/artifacts/UniversalProfile.json';
+import { ERC725, ERC725JSONSchema } from '@erc725/erc725.js';
+import LSP6Schema from '@erc725/erc725.js/schemas/LSP6KeyManager.json';
+import { getLuksoProvider } from '@/utils/provider';
+import {
+  DEFAULT_UP_CONTROLLER_PERMISSIONS,
+  DEFAULT_UP_URD_PERMISSIONS,
+  GRAVE_CONTROLLER_PERMISSIONS,
+} from '@/app/constants';
+import { ERC725YDataKeys, LSP1_TYPE_IDS } from '@lukso/lsp-smart-contracts';
+
+export const getChecksumAddress = (address: string | null) => {
+  // Check if the address is valid
+  if (!address || !ethers.utils.isAddress(address)) {
+    // Handle invalid address
+    return address;
+  }
+
+  // Convert to checksum address
+  return ethers.utils.getAddress(address);
+};
+
+export const hasOlderGraveDelegate = (
+  URDLsp7: string | null,
+  URDLsp8: string | null
+): string | null => {
+  const networkConfig = getNetworkConfig(
+    process.env.NEXT_PUBLIC_DEFAULT_NETWORK!
+  );
+  if (!URDLsp7 || !URDLsp8) return null;
+  const urd7 = getChecksumAddress(URDLsp7)!;
+  const urd8 = getChecksumAddress(URDLsp8)!;
+  let urd7Index = networkConfig.previousGraveForwarders.indexOf(urd7);
+  let urd8Index = networkConfig.previousGraveForwarders.indexOf(urd8);
+  if (urd7Index > -1 && urd8Index > -1 && urd7 === urd8) {
+    return networkConfig.previousGraveForwarders[urd7Index];
+  }
+  return null;
+};
+
+/**
```

Comment:
The functions below are copied from JoinGraveBtn 
I'll raise another PR to de-duplicate them